packages/xml-views-validation/src/validators/attributes/unknown-enum-value.ts

```diff
@@ -0,0 +1,57 @@
+import { XMLAttribute } from "@xml-tools/ast";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import {
+  flattenProperties,
+  xmlToFQN
+} from "@ui5-language-assistant/logic-utils";
+import { find, map } from "lodash";
+import { UnknownEnumValueIssue } from "../../../api";
+
+// TODO: Some logic here is very similar to: `enumSuggestions` in xml-view-completions,
+//  evaluate extracting the common parts
+export function validateUnknownEnumValue(
+  attribute: XMLAttribute,
+  model: UI5SemanticModel
+): UnknownEnumValueIssue[] {
+  const elementTagFqn = xmlToFQN(attribute.parent);
+  const ui5Class = model.classes[elementTagFqn];
+  if (ui5Class === undefined) {
+    return [];
+  }
+
+  const properties = flattenProperties(ui5Class);
+  const ui5Property = find(properties, ["name", attribute.key]);
+  const propType = ui5Property?.type;
+  if (propType?.kind !== "UI5Enum") {
+    return [];
+  }
+
+  const actualAttributeValue = attribute.value;
+  const actualAttributeValueToken = attribute.syntax.value;
+  if (
+    actualAttributeValue === null ||
+    actualAttributeValueToken === undefined
+  ) {
+    return [];
+  }
+
+  const possibleEnumValues = map(propType.fields, _ => _.name);
+  if (find(possibleEnumValues, _ => _ === actualAttributeValue) === undefined) {
+    const possibleValuesWithQuotes = map(possibleEnumValues, _ => `"${_}"`);
+    return [
+      {
+        kind: "UnknownEnumValue",
+        message: `Unknown enum value: ${
```

Comment:
Did you copy the message from FS?

Reply:
No. I will have a look

Reply:
I don't see that we had such a check in the past:
- https://github.wdf.sap.corp/DevX/xmlvalidator/tree/master/src/fioriAnalysis/rules

Reply:
Okay.
We should make a list of all the strings displayed to the user and review them with UA. Should probably create an activity so we don't forget to do it.

Reply:
good point, done:
- https://sapjira.wdf.sap.corp/browse/DEVXKINGSTON-937